components/cdc/src/observer.rs

```diff
@@ -177,20 +177,23 @@ impl RegionChangeObserver for CdcObserver {
         event: RegionChangeEvent,
         _: StateRole,
     ) {
-        if let RegionChangeEvent::Destroy = event {
-            let region_id = ctx.region().get_id();
-            if let Some(observe_id) = self.is_subscribed(region_id) {
-                // Unregister all downstreams.
-                let store_err = RaftStoreError::RegionNotFound(region_id);
-                let deregister = Deregister::Delegate {
-                    region_id,
-                    observe_id,
-                    err: CdcError::request(store_err.into()),
-                };
-                if let Err(e) = self.sched.schedule(Task::Deregister(deregister)) {
-                    error!("cdc schedule cdc task failed"; "error" => ?e);
+        match event {
+            RegionChangeEvent::Destroy | RegionChangeEvent::Update(_) => {
```

Comment:
It seems that not all region update but just `Split` & `CommitMerge` are required to deregister.

Besided, shall we handle the event of role change (i.e. become follower from leader).

And I think it would be better to return more proper error (e.g. `EpochNotMatch` when split or merge happen).

Reply:
Role changes are already handled in `on_role_change`. And here we can't know the correct `EpochNotMatch` information, so we can replace it with a `RegionNotFound`. For both 2 errors TiCDCs will retry them.